src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let averageAge = 0;
+  let died18 = [];
+
+  century === 18
+    ? died18 = people.filter(d => d.sex === 'm'
+&& Math.ceil(d.died / 100) === 18)
+    : died18 = people.filter(d => d.sex === 'm');
```

Comment:
What if century isn't 18, but 19? Then your filter will not behave in the right way, as it's programmed to detect men only born in 18 century. It should be able to filter out men from any century, if century is provided as a function argument.

```suggestion
  
  const men = people.filter(person => person.sex === 'm'); // here we have all men from people array
  
  const menFromCentury = men.filter(person => Math.ceil(person.died / 100) === century); // here we filter out all men, whose death date divided by 100 and rounded by Math.ceil() is equal to century, provided as a function argument.
  
  const menForCalculations = century ? menFromCentury : men; // here we choose what array will be used for further calculations, depending on if century was provided or not.

// Now you can use menForCalculations for your further operations.
```